### PR TITLE
mattermost-desktop: 4.1.2 -> 4.2.0

### PR DESCRIPTION
**Pull request by jokogr**
###### Motivation for this change

Mattermost Desktop update was pending since November 2018.

New version seems to be using GTK 3 and has a Windows security vulnerability fixed (I guess no worries for that). 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jokogr** [comment]
P.S.: I have also added my GPG key

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Hello @jokogr 

Could you please drop
```nix
phases = [ "unpackPhase" "installPhase" ];
```

and replace it with?
```nix
dontBuild = true;
dontConfigure = true;
```

We'd like to deprecate that.
See: https://github.com/NixOS/nixpkgs/issues/28910

Additionally, after inspecting the output I noticed that everything was
just copied to `$out` with no type of hierarchy.

Could you refactor the `installPhase` to not do that?

This can be as simple as coping everything to `$out/share/mattermost-desktop`
and then symlinking things from there.

**Comment by jokogr** [comment]
@worldofpeace thanks for the review!

I've refactored the phases to use `$out/share/mattermost-desktop`, dropped `lib` and added myself as a maintainer. I excluded the `patchPhase` as well, since shrinking the rpath for the binary makes it not working (I get a SIGTRAP).

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 1]
```diff
@@ -38,44 +38,52 @@ let
...
+      mkdir -p "$out/bin"
+      ln -s $out/share/mattermost-desktop/mattermost-desktop \
+        $out/bin/mattermost-desktop
 
-      patchShebangs $out/create_desktop_file.sh
-      $out/create_desktop_file.sh
+      patchShebangs $out/share/mattermost-desktop/create_desktop_file.sh
+      $out/share/mattermost-desktop/create_desktop_file.sh
+      rm $out/share/mattermost-desktop/create_desktop_file.sh
+      mkdir -p $out/share/applications
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix]:
```suggestion

      mkdir -p $out/share/applications
```

File: pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix [hunk 2]
```diff
@@ -38,44 +38,52 @@ let
...
+      mv ./Mattermost.desktop \
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/mattermost-desktop/default.nix]:
```suggestion
      mv Mattermost.desktop \
```

**Comment by jokogr** [comment]
@worldofpeace ready!

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**